src/Ocelot.Provider.Kubernetes/EndPointClientV1.cs

```diff
@@ -31,30 +27,7 @@ public async Task<EndpointsV1> GetAsync(string serviceName, string kubeNamespace
                 ServiceName = serviceName,
             });
 
-        var response = await Http.GetAsync(request, cancellationToken);
-
-        if (!response.IsSuccessStatusCode)
-        {
-            if (_logger.IsEnabled(LogLevel.Debug))
-            {
-                StatusV1 errorResponse = await response.ReadContentAsAsync<StatusV1>();
-
-                (string resourceKind, string resourceApiVersion) = KubeObjectV1.GetKubeKind<EndpointsV1>();
-                _logger.LogDebug("Failed to retrieve {ResourceApiVersion}/{ResourceKind} {ResourceName} in namespace {ResourceNamespace} ({HttpStatusCode}/{Status}/{StatusReason}): {StatusMessage}",
-                    resourceApiVersion,
-                    resourceKind,
-                    serviceName,
-                    kubeNamespace,
-                    response.StatusCode,
-                    errorResponse.Status,
-                    errorResponse.Reason,
-                    errorResponse.Message
-                );
-            }
-
-            return null;
-        }
-
-        return await response.ReadContentAsAsync<EndpointsV1>();
+        return await Http.GetAsync(request, cancellationToken)
+            .ReadContentAsObjectV1Async<EndpointsV1>(operationDescription: $"get {nameof(EndpointsV1)}");
```

Comment:
This wrapper method (part of KubeClient) will take care of deserialising the response from the server (if the status code does not indicate a success, it will attempt to deserialise the response body as a Kubernetes `StatusV1` and wrap that up in a `KubeApiException`.

Reply:
That's OK